app/android/runtime_client/src/org/xwalk/app/runtime/extension/ReflectionHelper.java

```diff
@@ -0,0 +1,265 @@
+package org.xwalk.app.runtime.extension;
+
+import android.util.Log;
+
+import java.lang.reflect.*;
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.Set;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+class ReflectionHelper {
+    private static final String TAG = "JsStubReflectHelper";
+    private Class<?> myClass;
+    private Map<String, MemberInfo> members = new HashMap<String, MemberInfo>();
+    private String[] eventList = null;
+    static Set<Class<?>> primitives = new HashSet<>();
+
+    public enum MemberType {
+        JS_METHOD,
+        JS_PROPERTY,
+    }
+
+    public class MemberInfo {
+        MemberType type;
+        boolean isWritable;
+        AccessibleObject accesser;
+    }
+
+    public ReflectionHelper(Class<?> clazz) {
+        myClass = clazz;
+        init();
+    }
+
+    void getMemberInfo(AccessibleObject[] accessers, MemberType type) {
+        for (AccessibleObject a : accessers) {
+
+            if (a.isAnnotationPresent(JsAPI.class)){
+                JsAPI mAnno = a.getAnnotation(JsAPI.class);
+                String name = ((Member) a).getName();
+
+                //Get eventList from propeties
+                if (type == MemberType.JS_PROPERTY && mAnno.isEventList()) {
+                    if (!((Field)a).getType().equals(String[].class)) {
+                        Log.w(TAG, "Invalid type for Supported JS event list" + name);
+                        continue;
+                    }
+                    try {
+                        // Event List should be a class property with "static"
+                        eventList = (String[])(((Field)a).get(null));
+                    } catch (IllegalArgumentException e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
+                    } catch (IllegalAccessException e) {
```

Comment:
I think we can write multiple catching in one line, already, but please double check.